webapp/channels/src/components/integrations/abstract_command.tsx

```diff
@@ -12,81 +11,66 @@ import * as Utils from 'utils/utils';
 import FormError from 'components/form_error';
 import SpinnerButton from 'components/spinner_button';
 import LocalizedInput from 'components/localized_input/localized_input';
+import ExternalLink from 'components/external_link';
 
 import {t} from 'utils/i18n';
-import ExternalLink from 'components/external_link';
+import {Command} from '@mattermost/types/integrations';
+import {Team} from '@mattermost/types/teams';
 
 const REQUEST_POST = 'P';
 const REQUEST_GET = 'G';
 
-export default class AbstractCommand extends React.PureComponent {
-    static propTypes = {
-
-        /**
-        * The current team
-        */
-        team: PropTypes.object.isRequired,
-
-        /**
-        * The header text to render, has id and defaultMessage
-        */
-        header: PropTypes.object.isRequired,
-
-        /**
-        * The footer text to render, has id and defaultMessage
-        */
-        footer: PropTypes.object.isRequired,
-
-        /**
-        * The spinner loading text to render, has id and defaultMessage
-        */
-        loading: PropTypes.object.isRequired,
-
-        /**
-         * Any extra component/node to render
-         */
-        renderExtra: PropTypes.node.isRequired,
-
-        /**
-        * The server error text after a failed action
-        */
-        serverError: PropTypes.string.isRequired,
-
-        /**
-        * The Command used to set the initial state
-        */
-        initialCommand: PropTypes.object,
-
-        /**
-        * The async function to run when the action button is pressed
-        */
-        action: PropTypes.func.isRequired,
-    };
+type Props = {
+    team: Team; // The current team
+    header: {id: string; defaultMessage: string}; // The header text to render, has id and defaultMessage
+    footer: {id: string; defaultMessage: string}; // The footer text to render, has id and defaultMessage
```

Comment:
Can this be MessageDescriptor from react-intl itself.

Reply:
yes, makes sense